elixir/apps/web/lib/web/live/settings/directory_sync.ex

```
@@ -319,6 +319,27 @@ defmodule Web.Settings.DirectorySync do
     end
   end
 
+  def handle_event("sync_directory", %{"id" => id, "type" => "okta"}, socket) do
+    case Oban.insert(Domain.Okta.Sync.new(%{"directory_id" => id})) do
+      {:ok, _job} ->
+        {:noreply, put_flash(socket, :info, "Directory sync has been queued successfully.")}
+
+      {:error, reason} ->
+        Logger.error("Failed to enqueue Okta sync job", id: id, reason: inspect(reason))
+        {:noreply, put_flash(socket, :error, "Failed to queue directory sync.")}
+    end
+  end
+
+  def handle_event("sync_directory", %{"id" => _id, "type" => type}, socket)
+      when type in ["entra", "google"] do
+    # TODO: Implement directory sync functionality for Entra and Google
+    {:noreply, put_flash(socket, :info, "Directory sync for #{type} coming soon.")}
+  end
+
+  def handle_event("sync_directory", %{"id" => _id, "type" => _type}, socket) do
+    {:noreply, put_flash(socket, :error, "Unknown directory type.")}
+  end
+
```

```suggestion
```

Think these cases can't be hit in practice

---

The `when type in ["entra", "google"]` should hopefully have their own implementations for handling the `sync_directory` event.  I'm curious why leaving a catch-all in would be a bad thing though?

---

> should hopefully have their own implementations

Yep they do, just right above.

> I'm curious why leaving a catch-all in would be a bad thing though?

I suppose it's not _necessarily_ a bad thing, but I don't believe it's possible to actually hit this path from the UI under any condition. I was thinking it's probably more clear if we let it crash and get a nice error notification about it rather than fail silently, because that would certainly be highly suspect.

---

Ok, I'll lean in to this "let it crash" thing 😄 

I would be interested in having a more in depth discussion, at some point, of where we should be shooting for in terms of the balance between "let it crash" and "catch any and all exceptions/errors".  Because we can still be "noisy" and not crash at the same time.

I'll remove this for now though.

---

> I would be interested in having a more in depth discussion, at some point, of where we should be shooting for in terms of the balance between "let it crash" and "catch any and all exceptions/errors". Because we can still be "noisy" and not crash at the same time.

I kind of think of this particular "let it crash" case as a runtime assertion. This "should never" happen. If it does, we need our assumptions / view of the universe adjusted.

Then I think there's "this isn't likely to happen but we cannot prove it won't". So I/O errors, race conditions, invalid data from an API. I think in these cases we can choose what to do depending on the context. If we're syncing and get an invalid API response, we might want to "let it crash" and start over so that when we reach the end of the sync process, we're sure we have a cohesive view of the provider's directory.

If we're in the replication consumer and crash, likely the same data will come back and bite us again, so we need to log and continue.